### PR TITLE
Add support for passing attributes

### PR DESCRIPTION
**Pull request by vjspranav**

### CODE REVIEW

File: Task.js [hunk 1]
```diff
@@ -184,6 +189,8 @@ class Task extends Unit {
 
       case ContentTypes.ASSESMENT:
         page_data.isAssesment = true;
+        page_data.content_type = "ASSESSMENT";
```
Comment (raj-vlabs) [inline Task.js:192]: Why do you need new content_types when this.content_type is already available?

File: plugin-config.testing.js [hunk 2]
```diff
@@ -1,10 +1,16 @@
...
 const config = [
   {
     id: "plugin-bug-report",
     scope: PluginScope.PAGE,
-    js_modules: ["https://vjspranav.github.io/vleads-bug-report/client/app.js"],
+    js_modules: [
+      "https://vjspranav.github.io/vleads-bug-report/client/src/bug-report.js",
+    ],
+    attributes: {
+      bug_options: JSON.stringify(bug_options),
```
Comment (raj-vlabs) [inline plugin-config.testing.js:12]: try to find better(more generic) names please.

File: templates/assets/css/vlabs-style.css [hunk 3]
```diff
@@ -156,6 +156,7 @@
   font-family: "Raleway", sans-serif;
   color: #000000;
   text-align: justify;
+  hyphens: auto;
```
Comment (raj-vlabs) [inline templates/assets/css/vlabs-style.css:159]: Is this change going in with this commit?

File: templates/pages/content.handlebars [hunk 4]
```diff
@@ -42,7 +43,16 @@
 	  </div>
           <div class="flex-column">
             <div id="plugin-bug-report"></div>
```
Comment (raj-vlabs) [inline templates/pages/content.handlebars:45]: do we still need this div?

File: templates/pages/content.handlebars [hunk 5]
```diff
@@ -42,7 +43,16 @@
...
-            <div id="plugin-rating"></div>
+            <bug-report
```
Comment (raj-vlabs) [inline templates/pages/content.handlebars:46]: Please name the component starting with vl- to reduce the possibilities of name collisions with other components

File: templates/pages/content.handlebars [hunk 6]
```diff
@@ -42,7 +43,16 @@
...
+              developer-institute="{{ collegeName }}"
+              context-info="Virtual Labs"
```
Comment (raj-vlabs) [inline templates/pages/content.handlebars:48]: context-info should have everything that is related to the context. Everything that is general should be independent attributes.
Think in terms of algodynamics, the attributes which would be applicable for both algodynamics and virtual labs should be attributes. The ones which would only appear in one of them, are part of context-info
Please discuss to understand what can be part of context and what can be generic attributes.

File: plugin-config.testing.js [hunk 7]
```diff
@@ -1,10 +1,16 @@
 const { PluginScope } = require("./Enums.js");
+const bug_options = require("./assets_plugins/json/bug-report-questions.js");
 
 const config = [
   {
     id: "plugin-bug-report",
```
Comment (raj-vlabs) [inline plugin-config.testing.js:6]: let's use the repo name as the plugin id

File: Task.js [hunk 8]
```diff
@@ -133,6 +134,8 @@ class Task extends Unit {
       isSimulation: false,
       isAssesment: false,
       assets_path: assets_path,
+      lab_data: lab_data,
+      exp_info: exp_info,
```
Comment (raj-vlabs) [inline Task.js:138]: Why do we need lab_data, exp_info and page_plugins in page_data?

**Review by raj-vlabs — CHANGES_REQUESTED** [review]

File: Task.js (hunk 1 above)
Comment (vjspranav) [reply Task.js:192]: Here sir, I need data in page_data so as to make it available in handlebars. Having a normal content type won't work. 
This gets passed as variable to handlebars so we can use it as attribute in compile time.

File: plugin-config.testing.js (hunk 2 above)
Comment (vjspranav) [reply plugin-config.testing.js:12]: for inside attributes, I was planning, to just loop over all attributes, and assign them to webcomponent, this allows us to define attributes in json if we need to, mainly for questions

File: templates/assets/css/vlabs-style.css (hunk 3 above)
Comment (vjspranav) [reply templates/assets/css/vlabs-style.css:159]: I didn't notice that, I'll remove it.

File: templates/pages/content.handlebars (hunk 4 above)
Comment (vjspranav) [reply templates/pages/content.handlebars:45]: No sir we don't

File: templates/pages/content.handlebars (hunk 5 above)
Comment (vjspranav) [reply templates/pages/content.handlebars:46]: sure sir, I'll update at both places.

File: Task.js (hunk 8 above)
Comment (vjspranav) [reply Task.js:138]: I needed them intially for other purposes,
right now I just need page_plugins, so I can pass attributes (like questions) at build time.

**Pull request closed without merging**